test/testdata/bound_proc.rb

```diff
@@ -0,0 +1,63 @@
+# typed: true
```

Comment:
Can you please copy the entire test case that I wrote in my branch into this PR?